cli/cli/Options/SaveToFileOption.cs

```diff
@@ -1,6 +1,73 @@
-﻿namespace cli;
+﻿using System.CommandLine;
 
-public class SaveToFileOption : ConfigurableOptionFlag
+namespace cli;
+
+public class SaveToFileOption
 {
-	public SaveToFileOption() : base("save-to-file", "Save login refresh token to file") { }
+	/// <summary>
+	/// Add the --save-to-file and --no-token-save options to a command.
+	/// The --save-to-file existed in 1.x, and was defaulted to FALSE.
+	///
+	/// In 2.0, --no-token-save was introduced, and the intention is that tokens
+	/// DO save by default. The new option exists as a way to OPT OUT. 
+	/// </summary>
+	/// <param name="command"></param>
+	/// <typeparam name="T"></typeparam>
+	public static void Bind<T>(AppCommand<T> command)
+		where T : CommandArgs, IArgsWithSaveToFile
+	{
+		const string saveToFileStr = "--save-to-file";
+		const string noTokenSaveStr = "--no-token-save";
+
+		const bool saveToFileDefault = true;
+		const bool noTokenSaveDefault = false;
+		
+		// included for legacy reasons. 
+		var saveToFileOption = new Option<bool>(saveToFileStr, 
+			
+			// Hey You, Watch out! This default value affects the algorithm in the parse handling! 
+			getDefaultValue: () => saveToFileDefault, 
+			description: "Save login refresh token to file");
+		
+		// this option is legacy, and we don't really want people using it anymore. 
+		saveToFileOption.IsHidden = true;
+		
+		var noTokenOption = new Option<bool>(noTokenSaveStr, 
+			
+			// Hey You, Watch out! This default value affects the algorithm in the parse handling! 

```

Comment:
Same as above. Please reference the algorithm explicitly so that readers can understand it without too much hunting.